File: Health-Declaration-Backend/src/main/java/com/healthdeclaration/entities/Token.java
package com.healthdeclaration.entities;

import com.healthdeclaration.util.enumeration.TokenType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "tbl_token")
public class Token {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String token;

    @Enumerated(EnumType.STRING)
    private TokenType tokenType;

    private boolean expired;

    private boolean revoked;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
}

File: Health-Declaration-Frontend/src/app/model/token.ts
import { TokenType } from "@angular/compiler";
import { User } from "./user.interface";

export class Token {
    id: number;
    token: string;
    tokenType: TokenType;
    expired: boolean;
    revoked: boolean;
    user: User;

    constructor(id: number, token: string, tokenType: TokenType, expired: boolean, revoked: boolean, user: User) {
        this.id = id;
        this.token = token;
        this.tokenType = tokenType;
        this.expired = expired;
        this.revoked = revoked;
        this.user = user;
    }
}